### DISCUSSION TITLE
Sync product-info templates?

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#6427] [Ideas]
As this feature is going into Bootstrap v3.7.0 (https://github.com/lat9/ZCA-Bootstrap-Template/pull/364), what's the feeling for combining the `tpl_{product-type-page}_display.php` files in the base Zen Cart?

Other than some base HTML `id` or `class` differences, they're all pretty much the same.  Granted, `product_music_info` is an outlier which is why I've broken out the display of a product's "details" as a separately customizable element as well as having an _optional_ additional information block.

**Comment by simon1066** [comment]
Personally, I make significant use of the different product types (exc. product free shipping) and have made a varying number of modifications to said template files. There may be other ways to achieve those changes but using separate template files was pretty straightforward.

**Reply by lat9** [reply]
Those changes don't prohibit heavy-customization of the different product types' information pages, but they _do_ make it easier to simply reuse `tpl_product_info_display.php` which cuts down on the base offering changes-required when a new feature (think `ask-a-question`) is introduced.

**Comment by drbyte** [comment]
I'm agreeable to it.

Are you thinking `3.0`?  Or `2.1` ?

**Reply by lat9** [reply]
2.1

**Reply by drbyte** [reply]
Yes. It'll make things simpler to maintain (and customize more consistently), and doesn't break anything if someone misses/skips one of the files during an upgrade.

**Comment by drbyte** [comment]
I haven't fully analyzed what you did in the Bootstrap template for this.
I know that "document_general" excludes a whole lot of things that the others "include". I guess if the core it's inheriting from just organically skips anything that has no data available for it, then it may be fine. What about when one intentionally wants something "blank" but not "skipped"?

**Reply by lat9** [reply]
As mentioned to @simon1066, sites have the ability to fully control any template rendering by just not pulling in `tpl_product_info_display.php`.

I did file-compares (at least on Bootstrap) and (other than good ol' `product_music`) the only differences in processing was what prefix was applied to various divs' `id=` attributes.  The templates just 'honor' the settings provided by their page's header_php.php and main_template_vars.php.

**Reply by drbyte** [reply]
Right. The modularity does make it much simpler.

**Comment by lat9** [comment]
Provided in zc210 via https://github.com/zencart/zencart/pull/6741.
